nio/client/async_client.py

```diff
@@ -715,7 +738,13 @@ def add_response_callback(
             full_state=full_state
         )
 
-        response = await self._send(SyncResponse, method, path)
+        response = await self._send(
+            SyncResponse,
+            method,
+            path,
+            # + 5: give server a chance to naturally return before we timeout
+            timeout = None if timeout is None else timeout / 1000 + 5,
```

Comment:
Again this is a bit extreme, letting the server have only 5 additional seconds is asking for trouble considering that Synapse and especially matrix.org can take their time.